test-resources-extensions/test-resources-extensions-core/src/main/java/io/micronaut/test/extensions/testresources/TestResourcesPropertiesFactory.java

```diff
@@ -55,40 +58,50 @@ public TestResourcesTestPropertyProvider(Class<?> testClass, Map<String, Object>
 
         @Override
         public Map<String, String> getProperties() {
-            TestResourcesProperties annotation = testClass.getAnnotation(TestResourcesProperties.class);
-            if (annotation != null) {
-                String[] requestedProperties = annotation.value();
-                var client = TestResourcesClientFactory.fromSystemProperties()
-                    .orElse(TestResourcesClientHolder.lazy());
-                var testResourcesConfig = properties.entrySet()
-                    .stream()
-                    .filter(e -> e.getKey().startsWith(TEST_RESOURCES_PROPERTY_PREFIX))
-                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-                Map<String, String> resolvedProperties = Stream.of(requestedProperties)
-                    .map(v -> new Object() {
-                        private final String key = v;
-                        private final String value = client.resolve(v, Map.of(), testResourcesConfig).orElse(null);
-                    })
-                    .filter(o -> o.value != null)
-                    .collect(Collectors.toMap(e -> e.key, e -> e.value));
+            TestResourcesProperties directAnnotation = testClass.getAnnotation(TestResourcesProperties.class);
+            if (directAnnotation != null) {
+                return extractAnnotationProperties(directAnnotation);
+            } else {
+                Optional<TestResourcesProperties> metaAnnotation = Arrays.stream(testClass.getAnnotations())
+                    .map(annotation -> annotation.annotationType().getAnnotation(TestResourcesProperties.class))
+                    .filter(Objects::nonNull)
+                    .findFirst();
+                return metaAnnotation.map(this::extractAnnotationProperties)
+                    .orElse(Map.of());
```

Comment:
Actual support for meta annotations, rest of the changes is just noise of moving the properties extraction code to `extractAnnotationProperties`